3rdparty/dxc/CMakeLists.txt

```diff
@@ -60,11 +60,29 @@ else()
 endif()
 
 set(DXC_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/build" CACHE INTERNAL "")
-execute_process(COMMAND "${CMAKE_COMMAND}" -S "${CMAKE_CURRENT_SOURCE_DIR}/dxc" -B "${DXC_BUILD_DIR}" -G "${CMAKE_GENERATOR}" "-Ax64" -T "${CMAKE_GENERATOR_TOOLSET}" ${NBL_DXC_CMAKE_OPTIONS}
+# execute_process(COMMAND "${CMAKE_COMMAND}" -C "${CMAKE_CURRENT_SOURCE_DIR}/dxc/cmake/caches/PredefinedParams.cmake" -S "${CMAKE_CURRENT_SOURCE_DIR}/dxc" -B "${DXC_BUILD_DIR}"-D CMAKE_BUILD_TYPE:STRING=Debug   ${NBL_DXC_CMAKE_OPTIONS}
```

Comment:
bring back msvc versions and add appropriate if-statement to determine what toolset you use